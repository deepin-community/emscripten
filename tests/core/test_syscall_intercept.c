#include <assert.h>
#include <string.h>
#include <stdio.h>

// Import the syscall under a separate name
__attribute__((import_module("env"), import_name("__syscall_getcwd")))
long __orig_getcwd(long buf, long size);


long __syscall_getcwd(long buf, long size) {
  printf("__syscall_getcwd intercepted\n");
  return __orig_getcwd(buf, size);
}

int main() {
  char cwd[1024];
  int rtn = __syscall_getcwd((long)cwd, sizeof(cwd));
  assert(rtn > 0);
  printf("cwd = %s\n", cwd);
  return 0;
}
