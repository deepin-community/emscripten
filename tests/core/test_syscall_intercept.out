__syscall_getcwd intercepted
cwd = /
